packages/fsapp/src/beta-app/modal/types.ts

```diff
@@ -10,7 +10,7 @@ export interface ModalService {
   dismissAllModals(): Promise<void>;
 }
 
-export interface ModalOptions extends RNModalOptions, ModalProps {
+export interface ModalOptions extends RNModalOptions, ModalProps, Options {
```

Comment:
Adding `Options` to this type will not make the options automatically apply.

This change would require changes to the modal.context.tsx so that it properly applies the options set here.
https://github.com/brandingbrand/flagship/blob/69e99e0e7841f116fdb839833783633b7ea38ce7/packages/fsapp/src/beta-app/modal/modal.context.tsx#L89-L92

Reply:
@wSedlacek,
I didn't add any new options.
I only fixed the typing.
Whatever passed in the `modal.options` now may include properties defined in `Options` from `react-native-navigation`.
The way it's passed to `Navigation.showModal` doesn't need to change.

Reply:
The `modal.options` get's passed to this `.showModal()` call.
```tsx
      Navigation.showModal({
          stack: {
            id: MODALS_STACK,
            children: [
              {
                component: {
                  id,
                  name: modal.definitionId,
                  options: {
                    modal: modal.options,
                    topBar: modal.topBarOptions
                  },
                  passProps: {
                    resolve,
                    reject,
                    ...props
                  }
                }
              }
            ]
          }
        });
      });
```

You will notice that `modal.options` is assigned to the `modal` property in the `options` record.
This property has the following type. [Source](https://wix.github.io/react-native-navigation/6.12.2/api/options-root#modal)

```ts
export interface ModalOptions {
    /**
     * Control wether this modal should be dismiss using swipe gesture when the modalPresentationStyle = 'pageSheet'
     * #### (iOS specific)
     */
    swipeToDismiss?: boolean;
}
```

So for something like `modalPresentationStyle` to be applied you would need something like
```tsx
      Navigation.showModal({
          stack: {
            id: MODALS_STACK,
            children: [
              {
                component: {
                  id,
                  name: modal.definitionId,
                  options: {
                    modal: modal.options,
                    topBar: modal.topBarOptions,
                    modalPresentationStyle: modal.options.modalPresentationStyle
                  },
                  passProps: {
                    resolve,
                    reject,
                    ...props
                  }
                }
              }
            ]
          }
        });
      });
```